Author: Simon Ruderich <simon@ruderich.org>
Description:
 Use build flags from environment (dpkg-buildflags). Necessary for hardening
 flags. Also strip -pie/-fPIE from CFLAGS/LDFLAGS when used to link libraries.
 It conflicts with -fPIC (Closes: #661393).

diff -Naurp keyutils.orig/Makefile keyutils/Makefile
--- keyutils.orig/Makefile	2012-04-16 09:36:50.340598037 +0000
+++ keyutils/Makefile	2012-04-16 09:37:24.176829938 +0000
@@ -1,5 +1,9 @@
-CPPFLAGS	:= -I.
-CFLAGS		:= -g -Wall -Werror
+CPPFLAGS	+= -I.
+CFLAGS		+= -g -Wall -Werror
+# Libraries can't be compiled with -pie/-fPIE. Strip it from CFLAGS/LDFLAGS if
+# used.
+CFLAGS_LIB	:= $(filter-out -fPIE,$(CFLAGS))
+LDFLAGS_LIB	:= $(filter-out -fPIE,$(filter-out -pie,$(LDFLAGS)))
 INSTALL		:= install
 DESTDIR		:=
 SPECFILE	:= keyutils.spec
@@ -118,10 +122,10 @@ $(SONAME): $(LIBNAME)
 LIBVERS := -shared -Wl,-soname,$(SONAME) -Wl,--version-script,version.lds
 
 $(LIBNAME): keyutils.os version.lds Makefile
-	$(CC) $(CFLAGS) -fPIC $(LDFLAGS) $(LIBVERS) -o $@ keyutils.os $(LIBLIBS)
+	$(CC) $(CFLAGS_LIB) -fPIC $(LDFLAGS_LIB) $(LIBVERS) -o $@ keyutils.os $(LIBLIBS)
 
 keyutils.os: keyutils.c keyutils.h Makefile
-	$(CC) $(CPPFLAGS) $(VCPPFLAGS) $(CFLAGS) -fPIC -o $@ -c $<
+	$(CC) $(CPPFLAGS) $(VCPPFLAGS) $(CFLAGS_LIB) -fPIC -o $@ -c $<
 
 ###############################################################################
 #
